### PR TITLE
Fix #1521 Auto Purge Cloudflare cache if Cache Everything Rule is set

### LINKED ISSUE #1521: Proposal: Purge Cloudflare cache automatically
For customers that use the Cache Everything feature in Cloudflare (i.e. they are caching the HTML) we could automatically purge that cache along with ours according to our usual triggers. 

Currently customers have to manually purge that cache when updating content.

Not sure if it's possible to automatically detect the cache everything setting when our Cloudflare integration is connected, but if not, it could be an option in the add-on UI.

### PR DESCRIPTION
**Pull request by crystinutzaa**
Auto purge Cloudflare cache if Cache Everything Rule is set

Based on:
3.4.2 WP Rocket Minor Version doc

### CODE REVIEW

File: inc/functions/cloudflare.php [hunk 1]
```diff
@@ -463,6 +463,70 @@ function rocket_purge_cloudflare() {
...
+ *
+ * @param WP_Post $post       The post object
+ * @param array   $purge_urls URLs cache files to remove
+ * @param string  $lang       The post language
+ *
+ * @throws Exception          If any error occurs when doing the API request.
+ * @return mixed Object|bool  true if the purge is successful, WP_Error otherwise
+ */
+function rocket_purge_cloudflare_by_url( $post, $purge_urls, $lang ) {
+	if ( is_wp_error( $GLOBALS['rocket_cloudflare'] ) ) {
```
[the remark this answers is not in the document]
Comment (crystinutzaa) [reply inc/functions/cloudflare.php]: Yes, it should be set in this file on line 133

File: inc/functions/cloudflare.php [hunk 2]
```diff
@@ -463,6 +463,70 @@ function rocket_purge_cloudflare() {
...
+		return $GLOBALS['rocket_cloudflare'];
+	}
+
+	try {
+		$cf_cache = new Cloudflare\Zone\Cache( $GLOBALS['rocket_cloudflare']->auth );
+		$cf_purge = $cf_cache->purge_files( $GLOBALS['rocket_cloudflare']->zone_id, $purge_urls );
+
+		if ( ! isset( $cf_purge->success ) || empty( $cf_purge->success ) ) {
```
[the remark this answers is not in the document]
Comment (crystinutzaa) [reply inc/functions/cloudflare.php]: This is how is used in all functions from Cloudflare: see `set_rocket_cloudflare_minify()` line 347 .
I taught there is a reason behind it ;)

File: inc/functions/cloudflare.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/functions/cloudflare.php]: Fine :)

**Comment by Tabrisrp** [comment]
@crystinutzaa You will have to resolve the conflict before being able to merge this 👍

**Pull request merged**